batch/batch/batch.py

```diff
@@ -95,38 +143,42 @@ def job_record_to_dict(record: Dict[str, Any], name: Optional[str]) -> JobListEn
     if cost_breakdown is not None:
         cost_breakdown = cost_breakdown_to_dict(json.loads(cost_breakdown))
 
-    return {
-        'batch_id': record['batch_id'],
-        'job_id': record['job_id'],
-        'name': name,
-        'user': record['user'],
-        'billing_project': record['billing_project'],
-        'state': record['state'],
-        'exit_code': exit_code,
-        'duration': duration,
-        'cost': coalesce(record.get('cost'), 0),
-        'msec_mcpu': record['msec_mcpu'],
-        'cost_breakdown': cost_breakdown,
-    }
-
-
-async def cancel_batch_in_db(db, batch_id):
+    return cast(
```

Comment:
This one is interesting, took the incoming refactored one, however it seems the cost_breakdown should be a variable not the record['cost_breakdown'].
Diff details:

`diff --git a/batch/batch/batch.py b/batch/batch/batch.py
index d7df00de1..cecd83a5f 100644
--- a/batch/batch/batch.py
+++ b/batch/batch/batch.py
@@ -48,8 +48,9 @@ def batch_record_to_dict(record: Dict[str, Any]) -> Dict[str, Any]:
         duration_ms = None
         duration = None
 
-    if record['cost_breakdown'] is not None:
-        record['cost_breakdown'] = cost_breakdown_to_dict(json.loads(record['cost_breakdown']))
+    cost_breakdown = record.get('cost_breakdown')
+    if cost_breakdown is not None:
+        cost_breakdown = cost_breakdown_to_dict(json.loads(cost_breakdown))
 
     batch_response = {
         'id': record['id'],
@@ -70,8 +71,8 @@ def batch_record_to_dict(record: Dict[str, Any]) -> Dict[str, Any]:
         'duration_ms': duration_ms,
         'duration': duration,
         'msec_mcpu': record['msec_mcpu'],
-        'cost': coalesce(record['cost'], 0),
-        'cost_breakdown': record['cost_breakdown'],
+        'cost': coalesce(record.get('cost'), 0),
+        'cost_breakdown': cost_breakdown,
     }
 
     attributes = json.loads(record['attributes'])
@@ -138,9 +139,39 @@ def job_record_to_dict(record: Dict[str, Any], name: Optional[str]) -> JobListEn
         exit_code = None
         duration = None
 
-    if record['cost_breakdown'] is not None:
-        record['cost_breakdown'] = cost_breakdown_to_dict(json.loads(record['cost_breakdown']))
+    cost_breakdown = record.get('cost_breakdown')
+    if cost_breakdown is not None:
+        cost_breakdown = cost_breakdown_to_dict(json.loads(cost_breakdown))
 
+<<<<<<< HEAD
+    return {
+        'batch_id': record['batch_id'],
+        'job_id': record['job_id'],
+        'name': name,
+        'user': record['user'],
+        'billing_project': record['billing_project'],
+        'state': record['state'],
+        'exit_code': exit_code,
+        'duration': duration,
+        'cost': coalesce(record.get('cost'), 0),
+        'msec_mcpu': record['msec_mcpu'],
+        'cost_breakdown': cost_breakdown,
+    }
+||||||| dc7fce0e3
+    return {
+        'batch_id': record['batch_id'],
+        'job_id': record['job_id'],
+        'name': name,
+        'user': record['user'],
+        'billing_project': record['billing_project'],
+        'state': record['state'],
+        'exit_code': exit_code,
+        'duration': duration,
+        'cost': coalesce(record['cost'], 0),
+        'msec_mcpu': record['msec_mcpu'],
+        'cost_breakdown': record['cost_breakdown'],
+    }
+=======
     return cast(
         JobListEntryV1Alpha,
         {
@@ -157,6 +188,7 @@ def job_record_to_dict(record: Dict[str, Any], name: Optional[str]) -> JobListEn
             'cost_breakdown': record['cost_breakdown'],
         },
     )
+>>>>>>> 13de4e6ff
`

Reply:
Yep, well done: we have local patches on `cost` (uses `.get()`) and `cost_breakdown` (some extra code making a variable). I got one of these wrong in my replay of this while checking it :smile: